sklearn/utils/validation.py

```diff
@@ -1025,27 +1025,27 @@ def _check_sample_weight(sample_weight, X, dtype=None):
     """
     n_samples = _num_samples(X)
 
-    if dtype is not None and dtype not in [np.float32, np.float64]:
+    if hasattr(sample_weight, "dtype"):
```

Comment:
This is useless I think. When this is an array we will return it directly

Reply:
It could be an array of strings, returning it immediately can lead to problems later.